site/content/how-to/monitoring/troubleshooting.md

```diff
@@ -65,54 +79,312 @@ LAST SEEN   TYPE      REASON              OBJECT
+
+```shell
+kubectl exec -it -n nginx-gateway  <ngf-pod-name> -c nginx /bin/sh
+```
+
 ##### Logs
 
 Logs from the NGINX Gateway Fabric control plane and data plane can contain information that isn't available to status or events. These can include errors in processing or passing traffic.
 
-To see logs for the control plane container:
+1. To see logs for the:
+
+- Control plane container
+
+```shell
+kubectl -n nginx-gateway  logs <ngf-pod-name> -c nginx-gateway
+```
+
+- Data plane container
+
+```shell
+kubectl -n nginx-gateway  logs <ngf-pod-name> -c nginx
+```
+
+1. To only see error logs for control plane and data plane containers:
+
+For _nginx-gateway_ container, you can `grep` for the word `error` or change the log level to `error` by following steps in [Modify logging levels](#modify-logging-levels). Once you have modified log levels, you can `grep` for the word `debug` to check debug logs for further investigation.
 
 ```shell
-kubectl -n nginx-gateway logs <ngf-pod-name> -c nginx-gateway
+kubectl -n nginx-gateway  logs <ngf-pod-name> -c nginx-gateway | grep error
+```
+
+For example, an error message when telemetry is not enabled for NGINX Plus installations:
+
+```text
+kubectl logs -n nginx-gateway nginx-gateway-nginx-gateway-fabric-77f8746996-j6z6v | grep error
+Defaulted container "nginx-gateway" out of: nginx-gateway, nginx
+{"level":"error","ts":"2024-06-13T18:22:16Z","logger":"usageReporter","msg":"Usage reporting must be enabled when using NGINX Plus; redeploy with usage reporting enabled","error":"usage reporting not enabled","stacktrace":"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static.createUsageWarningJob.func1\n\tgithub.com/nginxinc/nginx-gateway-fabric/internal/mode/static/manager.go:616\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:259\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:226\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:227\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:204\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:259\ngithub.com/nginxinc/nginx-gateway-fabric/internal/framework/runnables.(*CronJob).Start\n\tgithub.com/nginxinc/nginx-gateway-fabric/internal/framework/runnables/cronjob.go:53\nsigs.k8s.io/controller-runtime/pkg/manager.(*runnableGroup).reconcile.func1\n\tsigs.k8s.io/controller-runtime@v0.18.4/pkg/manager/runnable_group.go:226"}
 ```
 
-To see logs for the data plane container:
+For _nginx_ container you can `grep` for various [error](https://nginx.org/en/docs/ngx_core_module.html#error_log) log levels and they are all logged in brackets `[emerg]`:
```

Comment:
The tag corresponds to the error's level. So if the error is classified as `emerg` it will have the `[emerg]` tag, but if it's classified as `warn`, it will have the `[warn]` tag.